runelite-client/src/main/java/net/runelite/client/plugins/nightmarezone/NightmareZoneOverlay.java

```diff
@@ -148,6 +175,39 @@ public void removeAbsorptionCounter()
 		absorptionCounter = null;
 	}
 
+	public void resetPointsPerHour()
+	{
+		nmzSessionStartTime = null;
+		pointsPerHour = 0;
+	}
+
+	private int calculatePointsPerHour(int currentPoints)
+	{
+		Instant now = Instant.now();
+
+		if (nmzSessionStartTime == null)
+		{
+			nmzSessionStartTime = now;
+		}
+
+		if (lastPoints == currentPoints)
+		{
+			// prevent Points/Hour from flickering aggresively
```

Comment:
I'd kind of like to see this calculation happen in the plugin per game tick so that we can keep the logic consolidated there and avoid having this problem. That would also significantly simplify the overlay logic to simply building a panel with values fetched from the plugin.

Reply:
Pushed the requested changes.

I think I screwed up by squashing the commits, didn't I? Apologies for that.